core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -690,7 +690,7 @@ public List<ManifestFile> apply(TableMetadata base) {
         base.schema(), current != null ? current.dataManifests() : null);
     long minDataSequenceNumber = filtered.stream()
         .map(ManifestFile::minSequenceNumber)
-        .filter(seq -> seq > 0) // filter out unassigned sequence numbers in rewritten manifests
+        .filter(seq -> seq >= 0) // filter out unassigned sequence numbers in rewritten manifests
```

Comment:
Is this `filter` call even needed? If we need to keep results with sequence number 0, then it seems like we should just remove it. Everything should match that filter.

Reply:
some manifest file's minSequenceNumber is -1, if we remove this line, the minDataSequenceNumber will be -1 in some cases, and fail the validation in

Reply:
maybe we can change this line to

Reply:
That sounds good to me.